### DISCUSSION TITLE
How to assign discontinous numpy array to C++ array?

### DISCUSSION THREAD

**Discussion by kaixiongg** [pybind/pybind11#4322] [Q&A] [answered]
This issue occurs when I  tried to slice from 3d numpy array to 2d and then assign it to a C++ list. The result after slicing like np[:,:,1] is a discontinuous array (not in c or fortan order). I can't use std::copy(py::array.buf, my_continous_array_struct).

**Comment by kaixiongg** [comment, marked as answer]
I figured it out, we could call np.ascontiguousarray() inside pybind which enforce the array to be in c order(memory order).

**Reply by jiwaszki** [reply]
Take a look here as well: https://github.com/pybind/pybind11/discussions/4211#discussioncomment-3905115

You can use built-in capabilities of pybind to achieve the result.
